scilpy/image/tests/test_volume_math.py

```diff
@@ -572,6 +571,29 @@ def test_concatenate():
     assert_array_almost_equal(img_data_1.shape+(2,), output_data.shape)
 
 
+def test_correlation():
+    # Note. Not working on 2D data.
+    # Test 1: Data with background values
+    # Test 2: No background
+    # In both cases, compares a bunch of 0 with a bunch of 1, should be a
+    # perfect correlation.
+    img_data_1 = np.zeros((3, 3, 3)).astype(float)
```

Comment:
Here it wont work with an image full of zeros, because 0 is assume to be background (i.e. no information), to ensure there is no division by zero in np.corrcoef and no resulting NaN we add noise to background.

Therefore the correlation between an image full of zeros and full of ones is very poor (because one is 100% background), an image full of ones and twos would work (no background)

Reply:
@frheault After doing many tests...

Actually, result of corr is NaN whenever one patch has only the same value everywhere. So instead of adding random values, I managed possible NaN values: 

- 0 if at least one neighborhood was entirely containing background (deals with the case where we are in union of backgrounds, so no need to check for that anymore).
 - 1 if the voxel's neighborhoods are uniform in both images
 - 0 if the voxel's neighborhoods is uniform in one image, but not the other.

Tell me if that's ok with you.